packages/jest-mock/src/index.ts

```diff
@@ -1277,6 +1311,116 @@ export class ModuleMocker {
     return mock;
   }
 
+  mockProperty<
+    T extends object,
+    P extends PropertyLikeKeys<T>,
+    V extends Required<T>[P],
+  >(object: T, propertyKey: P, value: V): MockedProperty {
+    if (!object) {
+      throw new Error(
+        `mockProperty could not find an object on which to replace ${String(
+          propertyKey,
+        )}`,
+      );
+    }
+
+    if (!propertyKey) {
+      throw new Error('No property name supplied');
+    }
+
+    if (typeof object !== 'object') {
+      throw new Error(
+        `Cannot mock property on a primitive value; ${this._typeOf(
+          object,
+        )} given`,
+      );
+    }
+
+    let descriptor = Object.getOwnPropertyDescriptor(object, propertyKey);
+    let proto = Object.getPrototypeOf(object);
+    while (!descriptor && proto !== null) {
+      descriptor = Object.getOwnPropertyDescriptor(proto, propertyKey);
+      proto = Object.getPrototypeOf(proto);
+    }
+    if (!descriptor) {
+      throw new Error(`${String(propertyKey)} property does not exist`);
+    }
+    if (!descriptor.configurable) {
+      throw new Error(`${String(propertyKey)} is not declared configurable`);
+    }
+
+    if (this._isAlreadyMocked(object, propertyKey)) {
+      throw new Error(
+        `Cannot mock the ${String(
+          propertyKey,
+        )} property because it is already mocked`,
+      );
+    }
+
+    if (descriptor.get !== undefined) {
+      throw new Error(
+        `Cannot mock the ${String(
+          propertyKey,
+        )} property because it has a getter`,
+      );
+    }
+
+    if (descriptor.set !== undefined) {
+      throw new Error(
+        `Cannot mock the ${String(
+          propertyKey,
+        )} property because it has a setter`,
```

Comment:
Would we then use `spyOn` instead?

```suggestion
        )} property because it has a setter. Use \`jest.spyOn(object, '${propertyKey}', 'get').mockReturnValue(value)\` instead.`,
```